src/osdc/Objecter.cc

```diff
@@ -3435,6 +3435,7 @@ void Objecter::handle_osd_op_reply(MOSDOpReply *m)
     // new tid
     s->ops.erase(op->tid);
     op->tid = ++last_tid;
+    s->ops[op->tid] = op; // add op to ops again
```

Comment:
this is scaring. if we don't re-register the op with the new tid, the reply of this op would be taken as as a stray request, and the reply will be dropped on the floor. so the caller of op will be waiting in vain.